pymedphys/_dicom/anonymise/__init__.py

```diff
@@ -667,16 +670,16 @@ def _anonymise_tags(
 
 
 def _filter_identifying_keywords(
-    keywords_to_leave_unchanged, alternative_identifying_keywords=None
+    keywords_to_leave_unchanged, identifying_keywords=None
 ):
     r"""Removes DICOM keywords that the user desires to leave unchanged
     from the list of known DICOM identifying keywords and returns the
     resulting keyword list.
     """
-    if alternative_identifying_keywords is None:
-        keywords_filtered = list(IDENTIFYING_KEYWORDS)
+    if identifying_keywords is None:
+        keywords_filtered = get_default_identifying_keywords()
```

Comment:
I believe a `copy` might be needed here.

Reply:
See the following:

```python
In [1]: import functools

In [2]: @functools.lru_cache()
   ...: def a():
   ...:     return [1, 2, 3]
   ...:

In [3]: b = a()

In [4]: b
Out[4]: [1, 2, 3]

In [5]: b.append('oops')

In [6]: a()
Out[6]: [1, 2, 3, 'oops']
```

Essentially, running "keywords_filtered" once, could permanently remove them from the identifying list for that run through.

Reply:
I agree that the list needs to be safeguarded against mutation.
I've committed an alternative approach to achieving that, one that I think is a reasonable trade-off in performance/memory utilisation and minimising the need for the consumer to be "careful".

Reply:
Yup, awesome. I like that approach much better.